core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3322,49 +3369,55 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         }
       }
       if (syncPlan.toLoadMetadata()) {
-        loadMetadata = true;
+        AlluxioURI mountUri = new AlluxioURI(mMountTable.getMountPoint(inodePath.getUri()));
+        pathsToLoad.add(mountUri.getPath());
       }
       if (syncPlan.toSyncChildren() && inode instanceof InodeDirectory
           && syncDescendantType != DescendantType.NONE) {
         UfsStatus[] listStatus = ufs.listStatus(ufsUri.toString());
-        if (listStatus != null) {
-          InodeDirectory inodeDir = (InodeDirectory) inode;
-          // maps children name to inode
-          Map<String, Inode<?>> inodeChildren = new HashMap<>();
-          for (Inode<?> child : inodeDir.getChildren()) {
-            inodeChildren.put(child.getName(), child);
-          }
 
+        InodeDirectory inodeDir = (InodeDirectory) inode;
+        // maps children name to inode
+        Map<String, Inode<?>> inodeChildren = new HashMap<>();
+        for (Inode<?> child : inodeDir.getChildren()) {
+          inodeChildren.put(child.getName(), child);
+        }
+
+        if (listStatus != null) {
```

Comment:
Could you add a comment, something like: `// Iterate over UFS listing and process UFS children.` similar to the code block for inodes.

Reply:
done.